src/index.js

```diff
@@ -100,21 +116,49 @@ const createMarkdownShortcutsPlugin = (config = {}) => {
       }
       return 'not-handled';
     },
+    handleReturn(ev, { setEditorState, getEditorState }) {
+      const editorState = getEditorState();
+      const newEditorState = _handleReturn(editorState, ev);
+      if (editorState !== newEditorState) {
+        setEditorState(newEditorState);
+        return 'handled';
+      }
+      return 'not-handled';
+    },
     handleBeforeInput(character, { getEditorState, setEditorState }) {
       if (character !== ' ') {
         return 'not-handled';
       }
       const editorState = getEditorState();
-      let newEditorState = handleBlockType(editorState, character);
-      if (editorState === newEditorState) {
-        newEditorState = handleImage(editorState, character);
-      }
-      if (editorState === newEditorState) {
-        newEditorState = handleLink(editorState, character);
+      const newEditorState = _handleCharacter(editorState, character);
+      if (editorState !== newEditorState) {
+        setEditorState(newEditorState);
+        return 'handled';
       }
-      if (editorState === newEditorState) {
-        newEditorState = handleInlineStyle(editorState, character);
+      return 'not-handled';
+    },
+    handlePastedText(text, html, { getEditorState, setEditorState }) {
+      const editorState = getEditorState();
+      let newEditorState = editorState;
+      const buffer = [];
+      for (let i = 0; i < text.length; i++) {
+        if (INLINE_STYLE_CHARACTERS.indexOf(text[i]) >= 0) {
+          newEditorState = addText(newEditorState, buffer.join('') + text[i]);
+          newEditorState = _handleCharacter(newEditorState, text[i]);
+          buffer.length = 0;
```

Comment:
While typing Inline transformation were taking place we space is entered. 
But in case on pasting text.
`"_hello_ "` was transformed to _hello_. 
But `"_hello_"` was not being transformed. Thus while pasting INLINE_STYLE_CHARACTERS also checked.